### ISSUE TITLE
Is there a cache after deploying the Crawl4ai service on Docker? I

### ISSUE THREAD

**Issue by sz0811** [unclecode/crawl4ai#376]
Is there a cache after deploying the Crawl4ai service on Docker? I submitted different parameters twice, and the parameters of the large model were clearly incorrect. Surprisingly, I was able to return task_id and structured data normally?

**Comment by unclecode** [comment]
@sz0811 Yes, a cache exists by default based on the URL. However, if you want to get a fresh crawl every time, simply pass the cache mode and disable it. I will share the code snippet here for you.

```python
async def simple_crawl():
    print("\n--- Basic Usage ---")
    browser_config = BrowserConfig(headless=True)
    crawler_config = CrawlerRunConfig(
        cache_mode=CacheMode.BYPASS
    )
    
    async with AsyncWebCrawler(config=browser_config) as crawler:
        result = await crawler.arun(
            url="https://www.nbcnews.com/business",
            config=crawler_config
        )
        print(result.markdown[:500])
```

**Comment by sz0811** [comment]
Which file in the source code do I need to modify? Is it an asyncw_ebcrawler. py file? I made the changes and restarted Docker, but it doesn't seem to take effect yet

**Comment by unclecode** [comment]
@sz0811 You just simply set the cash mode to BYPASS, like below, if you are using docker. Does this make sense to you?

```python
request = {
    "urls": "https://www.nbcnews.com/business",
    "cache_mode": "bypass"
}
```

**Comment by sz0811** [comment]
After adding this parameter to the request, the cache is no longer returned, but why does it often report a request timeout? The URL can access a page, and the return parameters are as follows:
{"status":"completed","created_at":1735279247.1943,"result":{"url":" https://mzt.shanxi.gov.cn/zyfw/index.html ","html":"","success":false,"cleaned_html":null,"media":{},"links":{},"downloaded_files":null,"screenshot":null,"markdown":null,"markdown_v2":null,"fit_markdown":null,"fit_html":null,"extracted_content":null,"metadata":null,"error_message":"Failed on navigating ACS-GOTO :\nPage.goto: Timeout 60000ms exceeded.\nCall log:\nnavigating to \" https://mzt.shanxi.gov.cn/zyfw/index.html \", waiting until \"domcontentloaded\"\n","session_id":null,"response_headers":null,"status_code":null}}

**Comment by unclecode** [comment]
@sz0811 Timeouts may occur for different reasons. Sometimes, the page simply takes a long time to load, but you can change the default timeout. I suggest that before you use Docker during the development phase, try running it without Docker and set the headless option to false. This way, you can see what is happening in the browser, which will help you understand what went wrong. It's very helpful. Once you're confident about those cases, you can move to Docker. 

Anyway, when I checked the links you provided, I noticed that when I tried to open it in the browser, the page appeared but continued to load. Even after leaving it for five or six minutes, it was still loading. Crawl4ai after loading a url, waits for the network to become idle. The issue arises because this website does not load completely, causing idle issues. When I tested it, it took around five minutes for my browser to load the page completely. This means you need to just increase the timeout for waiting for the network to become idle, following the code below. 

```python
import asyncio
from crawl4ai import AsyncWebCrawler, BrowserConfig, CrawlerRunConfig, CacheMode
from crawl4ai.content_filter_strategy import PruningContentFilter
from crawl4ai.markdown_generation_strategy import DefaultMarkdownGenerator

async def main():
    # Configure the browser settings
    browser_config = BrowserConfig(headless=False, verbose=True)

    # Set run configurations, including cache mode and markdown generator
    crawl_config = CrawlerRunConfig(
        cache_mode=CacheMode.BYPASS,
        screenshot=True,
        scan_full_page=True,
        page_timeout=60 * 1000 * 2,
        
    )

    async with AsyncWebCrawler(config=browser_config) as crawler:
        result = await crawler.arun(
            url='https://mzt.shanxi.gov.cn/zyfw/index.html',
            config=crawl_config
            
        )
        if result.success:
            print("Raw Markdown Length:", len(result.markdown_v2.raw_markdown))
            print("Citations Markdown Length:", len(result.markdown_v2.markdown_with_citations))


if __name__ == "__main__":
    asyncio.run(main())
```

I will add this partial loading feature to my backlog for future planning. In situations like this, when most of the page loads, we can ignore it and continue to enable faster crawling. But right now, you can do this.

One last important thing is that I am working on a new way to use Docker models. It's totally different from the current approach. I want to let you know that the current method will be replaced by something much faster, and it has the benefit of not requiring communication with a web server or passing parameters like a dictionary. You will use the library directly as you do without Docker, but all the processes will happen in Docker. So stay tuned for that.

**Comment by sz0811** [comment]
I didn't use Docker to run it, but found the code you provided very useful. Thank you very much
But I have a question. I have a project developed in Java and I want to use this Crawl4ai tool. So, I deployed a Crawl4ai, and I need to call the deployed Crawl4ai Docker service through the Java side. However, based on the documentation, I only found two interfaces: IP: 11235/Crawl and IP: 11235/Task, and there is no rich documentation on how the interfaces pass parameters in different scenarios. What should I do? For example, in the Java project, I found that passing parameters like this did not take effect:
ip:11235/crawl 
request = {
"urls": " https://mzt.shanxi.gov.cn/zyfw/index.html ",
"page_timeout": "2*60000"
}
